jaxrs-clients/src/main/java/feign/CborDelegateDecoder.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package feign;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import com.google.common.net.HttpHeaders;
+import com.palantir.remoting2.jaxrs.feignimpl.HeaderAccessUtils;
+import feign.codec.DecodeException;
+import feign.codec.Decoder;
+import java.io.IOException;
+import java.io.PushbackInputStream;
+import java.lang.reflect.Type;
+import java.util.Collection;
+
+
+public final class CborDelegateDecoder implements Decoder {
```

Comment:
in the case of the decoder actually, we don't know at compile time... 

it's valid for a single endpoint on the server to return responses with different content types - your only way of telling is to look at the `Content-Type` header in the response